Fixes for build errors without devicetree support

wopu-ot opened the pull request:
Two minor fixes to avoid build errors on platforms that do not support devicetree.

wopu-ot committed:
device.h: Do not generate "okay" devicetree nodes if DT is unsupported

The macro DT_FOREACH_STATUS_OKAY_NODE is only available if devicetree
is supported. Using it for platforms that do not support devicetree
causes compilation errors.

Signed-off-by: Wolfgang Puffitsch <wopu@demant.com>

dcpleung commented:
I think it is better to change `source` to `osource` in `dts/Kconfig` so there is no need to create an empty file.

wopu-ot committed:
dts: Include Kconfig.dts as optional source

Avoid build errors for platforms that do not support devicetree and
hence do not generate a Kconfig.dts.

Signed-off-by: Wolfgang Puffitsch <wopu@demant.com>

mbolivar-nordic commented:
Fine, but I really think we need to have a conversation about whether DTS should remain optional.
